ODIN_II/SRC/adders.cpp

```diff
@@ -711,8 +711,12 @@ void split_adder(nnode_t* nodeo, int a, int b, int sizea, int sizeb, int cin, in
         connect_nodes(netlist->gnd_node, 0, node[0], (node[0]->num_input_pins - 1));
     }
 
+    // the sumout of the first dummy adder should connect to the next adder's cin
+    connect_nodes(node[0], 0, node[1], (node[1]->num_input_pins - 1));
+    connect_nodes(node[0], 1, node[1], (node[1]->num_input_pins - 1));
+
     //connect cout to next cin
-    for (i = 1; i < count; i++)
+    for (i = 2; i < count; i++)
```

Comment:
could you just remove the first node altogether.there is no reason to keep it(unless you can prove otherwise) it actually just breaks the expected result since the first adder is just connected to ground on both input except the carry.
just discard the first node and pin the carry where it belongs, can you also remove the last adder since it is not usefull